webserver/views/datasets.py

```diff
@@ -202,7 +203,7 @@ def view_json(id):
 
 
 @datasets_bp.route("/create", methods=("GET", "POST"))
-@login_required
+@api_login_required
 def create():
```

Comment:
This view does two things - on GET it returns HTML. This means that we should do a redirect if the user isn't logged in, but for a POST (sent by the react app) we should return json in the message if the user isn't logged in. This is difficult to do in a single method, so we should split this method into two, one to handle the GET and one for the POST